### PR TITLE
scripts: sbom: Report source and version information

### PR DESCRIPTION
**Pull request by doki-nordic**
Complete SBOM report should also contains information about a source of each package (module) and its version. This commit adds that information to the "west ncs-sbom" script.

Current preview of HTML report: https://doki-nordic.github.io/test-demo-docs/temp/west-sbom-sample.html

### CODE REVIEW

**Review by peknis — APPROVED** [review]

File: scripts/west_commands/sbom/git_info_detector.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+                                         return_error_code=True, allow_stderr=True)
+    if error_code != 0:
+        log.wrn(f'Directory "{relative_path}" does not provide valid git remote information.')
+        return None
+    remotes = split_lines(output)
+    if len(remotes) == 1:
+        return get_remote_url(absolute_path, remotes[0])
+    for remote in remotes:
+        if remote == 'origin':
+            return get_remote_url(absolute_path, remote)
```
Comment (maje-emb) [inline scripts/west_commands/sbom/git_info_detector.py:38]: It seems to me that `origin` will not always be the right remote.
Consider moving this after executing the `west list` command and checking whether sha exists in the remote.

**Review by maje-emb — COMMENTED** [review]
I think the documentation should be updated e.g. the  _Detectors_ section.

**Comment by carlescufi** [comment]
@doki-nordic can you address the review comments please?

File: scripts/west_commands/sbom/common.py [hunk 2]
```diff
@@ -50,19 +51,23 @@ def command_execute(*cmd_args: 'tuple[str|Path]', cwd: 'str|Path|None' = None,
...
             log.err(f'Command "{cmd_args[0]}" exited with error code {cp.returncode}')
             log.err(f'Arguments: { json.dumps(cmd_args) }, cwd: "{cwd}"')
             raise SbomException('Command execution error.')
         if return_path:
-            return Path(out_file.name)
+            result = Path(out_file.name)
         else:
-            return out_file.read()
+            result = out_file.read()
+        if return_error_code:
```
[the remark this answers is not in the document]
Comment (doki-nordic) [reply scripts/west_commands/sbom/common.py]: done

File: scripts/west_commands/sbom/git_info_detector.py (hunk 1 above)
Comment (doki-nordic) [reply scripts/west_commands/sbom/git_info_detector.py]: This makes sense, done.

**Review by maje-emb — APPROVED** [review]

**Comment by nordicjm** [comment]
Removing 2.2.0 label as requested

**Review by carlescufi — APPROVED** [review]

**Pull request merged**